lib/game_teams_repo.rb

```diff
@@ -0,0 +1,111 @@
+class GameTeamsRepo
+
+  def initialize(game_teams_path)
+    @game_teams = make_game_teams(game_teams_path)
+  end
+  
+  def make_game_teams(game_teams_path)
+    game_teams = []
+    CSV.foreach(game_teams_path, headers: true, header_converters: :symbol) do |row|
+      game_teams << GameTeams.new(row)
+    end
+    game_teams
+  end
+
+#REMEMBER TO REFACTOR THESE
+  def game_teams_by_team
+      @game_teams.group_by do |game|
+        game.team_id
+      end
+  end
+
+  def game_teams_by_away
+      @game_teams.group_by do |game|
+        game.team_id unless game.hoa == "home"
+      end
+  
+    end
+  
+  def game_teams_by_home
+    @game_teams.group_by do |game|
+      game.team_id unless game.hoa == "away"
+    end
+
+  end
+
+  def game_teams_by_coach
+    @game_teams.group_by do |game|
+      game.head_coach
+    end
+  end
+
+  def game_teams_by_team_id
+    game_set = {}
+    team_set = game_teams_by_team
+    team_set.map do |team, games|
+      game_set[team] = games.map do |game|
+        game.game_id
+      end
+    end
+    game_set
+  end
+
+  # def game_ids_by_season
+  # end
+
+  def game_team_by_season(game_ids, season_id)
+    @game_teams.find_all do |row|
+      game_ids[season_id].include?(row.game_id)
+    end
+  end
+
+  def best_offense
+    average_goals = {}
+    game_teams_by_team.map do |team , games|
```

Comment:
These are why we had crazy long test times. With most of these methods, it doesn't really make too much of a difference, but we should get into the habit of calling the method and storing the return value in a variable so the enumerable doesn't call the query method every time it iterates.